mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,392 @@
+        ###array, then make a dict with labels###
+        for i_chan in range(len(boxy_coords)):
+            boxy_coords[i_chan] = np.asarray(boxy_coords[i_chan],dtype=np.float64) 
+            
+        for i_chan in range(len(all_coords)):
+            all_coords[i_chan] = np.asarray(all_coords[i_chan],dtype=np.float64) 
+
+        all_chan_dict = dict(zip(all_labels,all_coords))
+
+         ###make our montage###
+        my_dig_montage = make_dig_montage(ch_pos=all_chan_dict,
+                                        coord_frame='unknown',
+                                        nasion = fiducial_coords[0],
+                                        lpa = fiducial_coords[1], 
+                                        rpa = fiducial_coords[2])
+        
+        ###create info structure###
+        info = create_info(boxy_labels, srate, ch_types='fnirs_raw')
+        ###add dig info###
+        ## this also applies a transform to the data into neuromag space based on fiducials
+        info.set_montage(my_dig_montage)
+
+        # Store channel, source, and detector locations
+        # The channel location is stored in the first 3 entries of loc.
+        # The source location is stored in the second 3 entries of loc.
+        # The detector location is stored in the third 3 entries of loc.
+        # Also encode the light frequency in the structure.
+       
+        ###place our coordinates and wavelengths for each channel###
+        # # These are all in actual 3d individual coordinates, so let's transform them to
+        # # the Neuromag head coordinate frame
+        trans = get_ras_to_neuromag_trans(fiducial_coords[0], 
+                                    fiducial_coords[1], 
+                                    fiducial_coords[2])
+        
+        for i_chan in range(len(boxy_labels)-1):
+            temp_chn = apply_trans(trans,boxy_coords[i_chan][0:3])
+            temp_src = apply_trans(trans,boxy_coords[i_chan][3:6])
+            temp_det = apply_trans(trans,boxy_coords[i_chan][6:9])
+            temp_other = np.asarray(boxy_coords[i_chan][9:],dtype=np.float64)
+            info['chs'][i_chan]['loc'] = test = np.concatenate((temp_chn, temp_src, 
+                                                                temp_det, temp_other),axis=0)
+        
+        info['chs'][-1]['loc'] = np.zeros((12,))        
+        
+        raw_extras = {'source_num': source_num,
+                     'detect_num': detect_num, 
+                     'start_line': start_line,
+                     'files': files}
+
+        print('Start Line: ', start_line)
+        print('End Line: ', end_line)
+        print('Original Difference: ', end_line-start_line)
+        first_samps = start_line
+        print('New first_samps: ', first_samps)
+        diff = end_line-start_line
+        #input file has rows for each source, output variable rearranges as columns and does not
+        last_samps = start_line + int(diff/source_num)-1 
```

Comment:
Use integer division
```suggestion
        last_samps = start_line + diff // source_num - 1
```